src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -227,6 +228,64 @@ export class MoveToLineEndTerminalAction extends BaseSendTextTerminalAction {
 	}
 }
 
+export class SendSequenceTerminalCommand extends Command {
+	public static readonly ID = TERMINAL_COMMAND_ID.SEND_SEQUENCE;
+	public static readonly LABEL = nls.localize('workbench.action.terminal.sendSequence', "Send Custom Sequence To Terminal");
+
+	public runCommand(accessor: ServicesAccessor, args: any): void {
+		const terminalInstance = accessor.get(ITerminalService).getActiveInstance();
+		if (!terminalInstance) {
+			return;
+		}
+		const text = args.text.split(' ');
+		var terminalText;
+		for (let i = 0; i < text.length; i++) {
+			terminalText = ' ' + prepareTerminaText(text[i]);
+			terminalInstance.sendText(terminalText, false);
+		}
+	}
+}
+
+function prepareTerminaText(text: string): string {
+	// Hexcodes and excape char
+	if (text.substring(0, 2).toLowerCase() === '\\x') {
+		// Valid hex with 2 digits
+		if (/^\\x[0-9a-f]{2}$/i.test(text)) {
+			return String.fromCharCode(parseInt(text.substring(2), 16));
+		}
+	}
+
+	// ctrl modifier with single char
+	if (text.indexOf('+') >= 0) {
+		const text1 = text.split('+');
+		if (text1[0].toLowerCase() === 'ctrl') {
+			if (text1[1].length === 1) {
+				return String.fromCharCode(text1[1].toUpperCase().charCodeAt(0) - 64);
+			}
+		}
+	}
+	// handle enter
+	if (text.toLowerCase() === 'enter') {
+		return '\n';
+	}
+
+	// Arrow keys
+	if (text.toLowerCase() === 'down') {
```

Comment:
I'm not sure these key strings should be handled specially, what if the user wants to send the text "down"?

Reply:
Should I remove all except `up` ?